File: payola/web/server/public/javascripts/runtime/s2js/runtime/scala/math/package.js
goog.provide('scala.math');
scala.math.E = Math.E;
scala.math.Pi = Math.PI;
scala.math.abs = function(x) {
var self = this;
return Math.abs(x);};
scala.math.acos = function(x) {
var self = this;
return Math.acos(x);};
scala.math.asin = function(x) {
var self = this;
return Math.asin(x);};
scala.math.atan = function(x) {
var self = this;
return Math.atan(x);};
scala.math.atan2 = function(x) {
var self = this;
return Math.atan2(x);};
scala.math.ceil = function(x) {
var self = this;
return Math.ceil(x);};
scala.math.cos = function(x) {
var self = this;
return Math.cos(x);};
scala.math.exp = function(x) {
var self = this;
return Math.exp(x);};
scala.math.floor = function(x) {
var self = this;
return Math.floor(x);};
scala.math.log = function(x) {
var self = this;
return Math.log(x);};
scala.math.max = function(x, y) {
var self = this;
return Math.max(x, y);};
scala.math.min = function(x, y) {
var self = this;
return Math.min(x, y);};
scala.math.pow = function(x, y) {
var self = this;
return Math.pow(x, y);};
scala.math.sqrt = function(x) {
var self = this;
return Math.sqrt(x);};
scala.math.random = function() {
var self = this;
return Math.random();};
scala.math.round = function(x) {
var self = this;
return Math.round(x);};
scala.math.sin = function(x) {
var self = this;
return Math.sin(x);};
scala.math.tan = function(x) {
var self = this;
return Math.tan(x);};
scala.math.signum = function(x) {
var self = this;
if(x > 0) { return 1; } else if(x == 0) { return 0; } else { return -1; }};
scala.math.__class__ = new s2js.Class('scala.math', []);

File: payola/s2js/runtime/client/src/main/scala/s2js/runtime/client/scala/math/package.scala
package s2js.runtime.client.scala.math

import s2js.compiler.javascript

object `package`
{
    @javascript("Math.E")
    val E: Double = 0

    @javascript("Math.PI")
    val Pi: Double = 0

    @javascript("return Math.abs(x);")
    def abs(x: AnyVal): Double = 0

    @javascript("return Math.acos(x);")
    def acos(x: AnyVal): Double = 0

    @javascript("return Math.asin(x);")
    def asin(x: AnyVal): Double = 0

    @javascript("return Math.atan(x);")
    def atan(x: AnyVal): Double = 0

    @javascript("return Math.atan2(x);")
    def atan2(x: AnyVal): Double = 0

    @javascript("return Math.ceil(x);")
    def ceil(x: AnyVal): Double = 0

    @javascript("return Math.cos(x);")
    def cos(x: AnyVal): Double = 0

    @javascript("return Math.exp(x);")
    def exp(x: AnyVal): Double = 0

    @javascript("return Math.floor(x);")
    def floor(x: AnyVal): Double = 0

    @javascript("return Math.log(x);")
    def log(x: AnyVal): Double = 0

    @javascript("return Math.max(x, y);")
    def max(x: AnyVal, y: AnyVal): Double = 0

    @javascript("return Math.min(x, y);")
    def min(x: AnyVal, y: AnyVal): Double = 0

    @javascript("return Math.pow(x, y);")
    def pow(x: AnyVal, y: AnyVal): Double = 0

    @javascript("return Math.sqrt(x);")
    def sqrt(x: AnyVal): Double = 0

    @javascript("return Math.random();")
    def random(): Double = 0

    @javascript("return Math.round(x);")
    def round(x: AnyVal): Double = 0

    @javascript("return Math.sin(x);")
    def sin(x: AnyVal): Double = 0

    @javascript("return Math.tan(x);")
    def tan(x: AnyVal): Double = 0

    @javascript("if(x > 0) { return 1; } else if(x == 0) { return 0; } else { return -1; }")
    def signum(x: AnyVal): Int = 0
}
